src/core/models.py

```diff
@@ -190,7 +239,7 @@ def rotate_passwords(self) -> bool:
     @property
     def quorum(self) -> str:
         """The current quorum encryption for the cluster."""
-        return self.relation_data.get("quorum", "")
+        return self.relation_data.get("quorum", "")  # pyright: ignore reportGeneralTypeIssues
```

Comment:
**suggestion** This is probably due to the somewhat slack typing we were forced by backward compatibility. If self.relation_data would take a `dict[str, str]` typing, we wouldn't probably need all these ignore. But instead of disabling pyright in a number of places, we can probably cast the `self.relation_data` into dict[str, str] in a single place ([here](https://github.com/canonical/zookeeper-operator/pull/129/files#diff-e46c88bb687d77b94764a732b3e08a0a7bd544a64e606429c0b787c13e1c226aR58)). That line would probably require ignore, but it is one line instead of many.

Reply:
`relation_data` has a `UserDict` type.

Reply:
https://github.com/canonical/data-platform-libs/pull/153

Reply:
Kafka/ZK does not need to support 3.8, so this isn't an issue.

Reply:
I responded to

>  If self.relation_data would take a dict[str, str] typing,

Now this is not ensured by the lib where that object is currently coming from.

Reply:
pyright ignore remove